cms/src/main/java/uk/nhs/digital/apispecs/swagger/IsAnyTrueHelper.java

```diff
@@ -0,0 +1,39 @@
+package uk.nhs.digital.apispecs.swagger;
+
+import com.github.jknack.handlebars.Helper;
+import com.github.jknack.handlebars.Options;
+import org.apache.commons.lang.Validate;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class IsAnyTrueHelper implements Helper<String> {
+
+    public static final IsAnyTrueHelper INSTANCE = new IsAnyTrueHelper();
+
+    public static final String NAME = "isAnyTrue";
+
+    @Override public Object apply(final String items, final Options options) throws IOException {
+
+        Validate.notEmpty(options.params,"One or more input parameters");
+        Validate.noNullElements(options.params,"One or more input parameters");
+        Validate.allElementsOfType(Arrays.asList(options.params), Boolean.class,"Required parameter type is Boolean");
```

Comment:
I'm afraid that in the current form this message is confusing and, potentially, misleading. 

This validation requires that all arguments are of type Boolean and fails if even one is not, and the error message should state it explicitly if we are using it at all.

As is, it could be easily misinterpreted as saying 'the method requires a non-boolean parameter but boolean was provided.

Yes, looking at the stack trace to see which validation method the exception came from does add clarifying context but if we require people to look at the stack trace to understand the problem, why use error messages at all?

Reply:
Fixed